File: BinarySearch_recursive.java
public class BinarySearch_recursive {



public static int binarysearch(int[] array, int left, int right, int key){
        int middle = (left + right) /2;

        if(array[middle] == key){

          return middle;
        }
        else if(array[middle] > key){
          return binarysearch(array, left, middle, key);


        }
        return binarysearch(array, middle, right, key);







}

public static void main(String[] args){

        int[] array = {2, 3, 4, 5, 6, 7, 8, 9};
        int key = 8;

        System.out.println(binarysearch(array, 0, array.length, key));


}

}

File: Searching/BinarySearch_recursive.py
def BinarySearch_recursive(array, left, right, key):

    if right >= left:

        middle = (left + right) // 2

        if array[middle] == key:
            return middle

        elif array[middle] > key:
            return BinarySearch_recursive(array, left, middle, key)

        else:
            return BinarySearch_recursive(array, middle, right, key)

    else:
        return -1


array = [ 2, 3, 4, 10, 40]
key = 10

result = BinarySearch_recursive(array, 0, len(array)-1, key)

print("Index is ", str(result))
